core/chain_makers.go

```diff
@@ -408,6 +413,19 @@ func GenerateVerkleChain(config *params.ChainConfig, parent *types.Block, engine
 				panic(err)
 			}
 
+			// Get the keys collected in the witness in the witness.
+			computedKeys := statedb.Witness().Keys()
+			sort.Slice(computedKeys, func(i, j int) bool { return bytes.Compare(computedKeys[i], computedKeys[j]) < 0 })
+
+			// Get the pre-state values from the database.
+			computedPreStateValues := make([][]byte, len(computedKeys))
+			for i := range computedKeys {
+				computedPreStateValues[i], err = preStateTree.GetWithHashedKey(computedKeys[i])
+				if err != nil {
+					panic(err)
+				}
+			}
```

Comment:
Nothing weird.

Reply:
nothing weird except the same work is more or less already done in `GetProofItems`. But yeah, I don't really want to expose it all and make the interface even more encumbered while we figure out what the best approach to do this is.